### ISSUE TITLE
Seeing a "Permission denied - /root" on server

### ISSUE THREAD

**Issue by psbanka** [dcsobral/puppet-users#7]
When attempting to deploy some user accounts, I am running into a server-side error:

err: Could not retrieve catalog from remote server: Error 400 on SERVER: Permission denied - /root at /etc/puppet/modules/users/manifests/definitions/useraccount.pp:65 on node server3.nisdev.foo.com

Any ideas?

**Comment by dcsobral** [comment]
Curious, because that line doesn't actually do anything, just defines a
string pointing to a directory that might be checked in the future.

The directory is "/etc/puppet/files/users/home/managed/host/${username}
.$fqdn", and if you look at the following lines, you'll see a bunch of
similar directories. Try making sure that
/etc/puppet/files/users/home/{managed,default}/{host,domain,env,user} are
directories with read and execute permissions for all, as well as all
parent directories.

Also check whether /etc/puppet/modules/users/scripts/findDirs.sh can be
executed by anyone, and try passing those directories to it to see if works
or not.

On Thu, Aug 16, 2012 at 6:04 PM, Peter Banka notifications@github.comwrote:

> When attempting to deploy some user accounts, I am running into a
> server-side error:
> 
> {code}
> err: Could not retrieve catalog from remote server: Error 400 on SERVER:
> Permission denied - /root at
> /etc/puppet/modules/users/manifests/definitions/useraccount.pp:65 on node
> server3.nisdev.foo.com
> {code}
> 
> Any ideas?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/dcsobral/puppet-users/issues/7.

## 

Daniel C. Sobral

I travel to the future all the time.

**Comment by psbanka** [comment]
Thanks for your reply.

I started experimented quite a bit with commenting sections of the
users/manifests/definitions/useraccount.pp out (looking for anything with
'root' in it) , and got the errors to stop occurring. Then I began
experimenting with adding files in to the users/files/home/default
directories.

As I got further along, I eventually uncommented more and more sections of
the original (useraccount.pp) file until it was back to its pristine state
and no longer throwing errors. I'm fairly puzzled as to where the error
originated now, and I wonder if perhaps an error was being thrown because
the directory tree was empty?

Anyway, it seems to be working find now.

I like your module quite a bit!

--Peter
